### PR TITLE
Improve commitment test helpers

### PR DESCRIPTION
**Pull request by andrewgordstewart**
See commit messages and comments.

### CODE REVIEW

File: packages/wallet/src/redux/__tests__/helpers.ts [hunk 1]
```diff
@@ -79,51 +79,114 @@ export const expectThisMessage = (state: SideEffectState, messageType: string) =
...
-  expectSideEffect('messageOutbox', state, item => {
-    expect(item.messagePayload.type).toEqual(messageType);
-    expect(item.messagePayload.signedCommitment.commitment).toMatchObject(c);
-  });
-};
 
-export const expectThisMessageAndTheseCommitmentsSent = (
+type PartialCommitments = Array<{ commitment: Partial<Commitment>; signature?: string }>;
+
+export function transformPartialCommitmentToObjectContainint(sc: {
```
Comment (geoknee) [inline packages/wallet/src/redux/__tests__/helpers.ts]: Typo here Containint -> Containing (presumably).

Could this function be better named?. Unless I misunderstand, the input type is not a `PartialCommitment` (it is an unnamed type which has a property with type `PartialCommitment`), and it does not return an `Object` (it returns a jest matcher).

Comment (andrewgordstewart) [reply packages/wallet/src/redux/__tests__/helpers.ts]: a Jest matcher is an object!

Comment (andrewgordstewart) [reply packages/wallet/src/redux/__tests__/helpers.ts]: I think it's too wordy to say `transformFromMaybeSignedPartialCommitmentToJestMatcher`. I would be more inclined to say `transformCommitmentToMatcher`. There's no questioning what kind of object it takes -- it's right there in the signature!

File: packages/wallet/src/redux/__tests__/helpers.ts [hunk 2]
```diff
@@ -74,51 +79,118 @@ export const expectThisMessage = (state: SideEffectState, messageType: string) =
...
-    expect(item.messagePayload.signedCommitment.commitment).toMatchObject(c);
-  });
-};
 
-export const expectThisMessageAndTheseCommitmentsSent = (
+type PartialCommitments = Array<{ commitment: Partial<Commitment>; signature?: string }>;
+
+function transformCommitmentToMatcher(sc: { commitment: Partial<Commitment>; signature?: string }) {
+  if (sc.signature) {
+    return expect.objectContaining({ signature: sc.signature });
```
Comment (kerzhner) [inline packages/wallet/src/redux/__tests__/helpers.ts]: Why does the expected object not contain a commitment property?

File: packages/wallet/src/redux/__tests__/helpers.ts [hunk 3]
```diff
@@ -74,51 +79,118 @@ export const expectThisMessage = (state: SideEffectState, messageType: string) =
...
+      // We try to help the developer by using expect(foo).toMatchObject(bar)
+      // The errors are much more useful in this case, but will be deceiving in the case when
+      // multiple messages are queued.
+
+      // To help with debugging, you can change the idx variable when running tests to 'search'
+      // for the correct commitment
 
-export const expectThisCommitmentSent = (state: SideEffectState, c: Partial<Commitment>) => {
-  expectThisMessageAndCommitmentSent(state, c, 'WALLET.COMMON.COMMITMENT_RECEIVED');
+      expect(messageOutbox[idx]).toMatchObject({
```
Comment (kerzhner) [inline packages/wallet/src/redux/__tests__/helpers.ts]: What about a `console.warn` message that points the developer to this code. Something like: 'Make sure to understand the matcher before diving into debugging'?

File: packages/wallet/src/redux/__tests__/helpers.ts [hunk 4]
```diff
@@ -74,51 +79,118 @@ export const expectThisMessage = (state: SideEffectState, messageType: string) =
...
+        },
+      });
+    } else {
+      throw err;
+    }
+  }
 };
-type PartialCommitments = Array<{ commitment: Partial<Commitment>; signature?: string }>;
 
 export const expectTheseCommitmentsSent = (
```
Comment (kerzhner) [inline packages/wallet/src/redux/__tests__/helpers.ts]: Should this use `expectThisCommitmentSent`?

File: packages/wallet/src/redux/__tests__/helpers.ts [hunk 5]
```diff
@@ -74,51 +79,118 @@ export const expectThisMessage = (state: SideEffectState, messageType: string) =
...
+        },
+      });
+    } else {
+      throw err;
+    }
+  }
 };
 
+function getOutboxState(state: SideEffectState, outboxBranch: 'messageOutbox'): MessageOutbox {
+  if ('sideEffects' in state && state.sideEffects && state.sideEffects[outboxBranch]) {
```
Comment (kerzhner) [inline packages/wallet/src/redux/__tests__/helpers.ts]: Why is `'sideEffects' in state` needed?

File: packages/wallet/src/redux/__tests__/helpers.ts (hunk 2 above)
Comment (andrewgordstewart) [reply packages/wallet/src/redux/__tests__/helpers.ts]: Good catch!

File: packages/wallet/src/redux/__tests__/helpers.ts (hunk 5 above)
Comment (andrewgordstewart) [reply packages/wallet/src/redux/__tests__/helpers.ts]: `SideEffectState` is a union type, where not all members have the `sideEffect` property. So, without, the compiler is unaware that `state` can have a `sideEffect` property.

File: packages/wallet/src/redux/__tests__/helpers.ts (hunk 4 above)
Comment (andrewgordstewart) [reply packages/wallet/src/redux/__tests__/helpers.ts]: No, because `expectThisCommitmentSent` is about sending `COMMITMENT_RECEIVED`, which only works for two-party channels.

In an n-party channel, we are currently sending the whole round of commitments, as this is the only thing that allows "forward-only" communication to work. This is a different action: `COMMITMENTS_RECEIVED`

Comment (kerzhner) [reply packages/wallet/src/redux/__tests__/helpers.ts]: Gotcha. It's unfortunate that most of the functionality between the `expectThisCommitmentSent` and `expectTheseCommitmentsSent` is copy-pasted. But this is test code after all...

Comment (andrewgordstewart) [reply packages/wallet/src/redux/__tests__/helpers.ts]: Yes, I thought about trying to dry it out a bit, but deemed it not worth the effort.
